### PR TITLE
Ldap exceptions

### PR DESCRIPTION
**Pull request by geoperez**

### CODE REVIEW

File: test/Unosquare.Swan.Test/LdapTest.cs [hunk 1]
```diff
@@ -1,61 +1,145 @@
...
+        private const string ldapServer = "ldap.forumsys.com";
 
-        [Test]
-        public async Task BindTest()
+        [TestFixture]
+        public class Bind : LdapTest
         {
-            var cn = new LdapConnection();
+            [Test]
+            public async Task UseValidCredentials()
```
Comment (geoperez) [inline test/Unosquare.Swan.Test/LdapTest.cs]: Is there a way to validate the result from bind?

File: test/Unosquare.Swan.Test/LdapTest.cs [hunk 2]
```diff
@@ -1,61 +1,145 @@
...
         }
 
-        [Test]
-        public async Task SearchTest()
+        [TestFixture]
+        public class Connect : LdapTest
         {
-            var cn = new LdapConnection();
+            [Test]
+            public async Task UseValidHostPort()
```
Comment (geoperez) [inline test/Unosquare.Swan.Test/LdapTest.cs]: UsingValidHost_RetursTrue

File: test/Unosquare.Swan.Test/LdapTest.cs [hunk 3]
```diff
@@ -1,61 +1,145 @@
...
+                }
+
+                Assert.AreNotEqual(lsc.Count, 0);
+                cn.Disconnect();
+            }
 
-            cn.Disconnect();
-            Assert.IsFalse(cn.Connected);
+            [Test]
+            public void UsingInvalidDN()
```
Comment (geoperez) [inline test/Unosquare.Swan.Test/LdapTest.cs]: UsingInvalidDn_LdapExceptionThrown

File: test/Unosquare.Swan.Test/LdapTest.cs (hunk 2 above)
Comment (Alexey-Unosquare) [reply test/Unosquare.Swan.Test/LdapTest.cs]: or just ValidHost_ReturnsTrue

File: test/Unosquare.Swan.Test/LdapTest.cs [hunk 4]
```diff
@@ -1,61 +1,145 @@
...
+                await cn.Connect(ldapServer, 389);
+                Assert.IsTrue(cn.Connected);
+                cn.Disconnect();
+            }
 
-            await cn.Connect("ldap.forumsys.com", 389);
-            await cn.Bind("uid=riemann,dc=example,dc=com", "password");
-            var lsc = await cn.Search("ou=scientists,dc=example,dc=com", LdapConnection.ScopeSub);
+            [Test]
+            public void UseInvalidHost()
```
Comment (Alexey-Unosquare) [inline test/Unosquare.Swan.Test/LdapTest.cs]: InvalidHost_ReturnsThrowsException

File: test/Unosquare.Swan.Test/LdapTest.cs [hunk 5]
```diff
@@ -1,61 +1,145 @@
...
+                    await cn.Connect("ldap.forumsys", 388);
+                    await cn.Bind("uid=riemann,dc=example,dc=com", "password");
+                    cn.Disconnect();
+                });
             }
 
-            Assert.AreNotEqual(lsc.Count, 0);
-            Assert.IsTrue(lsc.HasMore());
+            [Test]
+            public async Task Disconnect()
```
Comment (Alexey-Unosquare) [inline test/Unosquare.Swan.Test/LdapTest.cs]: void methods....maybe Connected_ResetsConnection()?

File: test/Unosquare.Swan.Test/LdapTest.cs [hunk 6]
```diff
@@ -1,61 +1,146 @@
...
+                await cn.Connect(ldapServer, 389);
+                Assert.IsTrue(cn.Connected);
+                cn.Disconnect();
+            }
 
-            await cn.Connect("ldap.forumsys.com", 389);
-            await cn.Bind("uid=riemann,dc=example,dc=com", "password");
-            var lsc = await cn.Search("ou=scientists,dc=example,dc=com", LdapConnection.ScopeSub);
+            [Test]
+            public void InvalidHost_SocketExceptionThrown()
```
Comment (Alexey-Unosquare) [inline test/Unosquare.Swan.Test/LdapTest.cs]: better _Throws...

File: test/Unosquare.Swan.Test/LdapTest.cs [hunk 7]
```diff
@@ -1,61 +1,146 @@
...
+                }
+
+                Assert.AreNotEqual(lsc.Count, 0);
+                cn.Disconnect();
+            }
 
-            cn.Disconnect();
-            Assert.IsFalse(cn.Connected);
+            [Test]
+            public void UsingInvalidDN_LdapExceptionThrown()
```
Comment (Alexey-Unosquare) [inline test/Unosquare.Swan.Test/LdapTest.cs]: InvalidDN_ThrowsLdapException

**Pull request merged**